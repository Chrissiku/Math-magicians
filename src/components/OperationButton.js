/* eslint-disable react/prop-types */
/* eslint-disable react/prefer-stateless-function */
/* eslint-disable react/destructuring-assignment */
import React from 'react';

class OperationButton extends React.Component {
  render() {
    // eslint-disable-next-line react/prop-types
    return (
      <button className="orange-color" type="button">
        {this.props.operation}
      </button>
    );
  }
}

export default OperationButton;
