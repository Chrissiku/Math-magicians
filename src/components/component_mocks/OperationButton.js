/* eslint-disable react/prop-types */
/* eslint-disable react/prefer-stateless-function */
/* eslint-disable react/destructuring-assignment */
import React from 'react';

class OperationButton extends React.Component {
  render() {
    // eslint-disable-next-line react/prop-types
    return (
      <button className="orange-color" type="button" onClick={this.props.e}>
        {this.props.name}
      </button>
    );
  }
}

export default OperationButton;
