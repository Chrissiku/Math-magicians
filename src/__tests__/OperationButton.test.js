/* eslint-disable react/jsx-props-no-spreading */
import renderer from 'react-test-renderer';
import OperationButton from '../components/component_mocks/OperationButton';

const props = { name: 'Chris', e: jest.fn() };
test('Test Operation button component', () => {
  const TREE = renderer.create(<OperationButton {...props} />).toJSON();
  expect(TREE).toMatchSnapshot();
});
