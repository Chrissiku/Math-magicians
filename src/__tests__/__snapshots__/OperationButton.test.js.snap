// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Operation button component 1`] = `
<button
  className="orange-color"
  onClick={[MockFunction]}
  type="button"
>
  Chris
</button>
`;
